cinder/volume/volume_types.py

```diff
@@ -375,3 +378,38 @@ def _get_encryption(enc):
     enc1_filtered = _get_encryption(enc1) if enc1 else None
     enc2_filtered = _get_encryption(enc2) if enc2 else None
     return enc1_filtered != enc2_filtered
+
+
+def provision_filter_on_size(context, volume_type, size):
+    """This function filters volume provisioning requests on size limits.
+
+    If a volume type has provisioning size min/max set, this filter
+    will ensure that the volume size requested is within the size
+    limits specified in the volume type.
+    """
+
+    if not volume_type:
+        volume_type = get_default_volume_type()
```

Comment:
I find it rather unexpected that this functions enforces the caller setting a `volume_type` but then doesn't fail if it's set to `None` but uses the default type instead. It's not visible from the parameters to me, that this function might change `volume_type` to a default (I would expect it if `volume_type=None`), and it's also not stated in the function comment.

Reply:
This is an artifact of cinder master branch where cinder enforces a default volume type to be set on a volume at creation time.   I can remove this for queens/our branch since cinder/queens doesn't enforce it.  This code already does a if volume_type right after this line, so this function would turn into a noop pass.